athena-gcs/src/main/java/com/amazonaws/athena/connectors/gcs/GcsUtil.java

```diff
@@ -0,0 +1,164 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.amazonaws.athena.connectors.gcs;
+
+import com.amazonaws.athena.connector.lambda.domain.TableName;
+import com.amazonaws.athena.connector.lambda.security.CachableSecretsManager;
+import com.amazonaws.athena.connectors.gcs.common.StorageSplit;
+import com.amazonaws.services.glue.AWSGlue;
+import com.amazonaws.services.glue.model.GetTableResult;
+import com.amazonaws.services.glue.model.Table;
+import com.amazonaws.services.secretsmanager.AWSSecretsManagerClientBuilder;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.arrow.vector.types.Types;
+import org.apache.arrow.vector.types.pojo.Field;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.Arrays;
+import java.util.List;
+
+import static com.amazonaws.athena.connectors.gcs.GcsConstants.GCS_SECRET_KEY_ENV_VAR;
+import static com.amazonaws.athena.connectors.gcs.GcsConstants.GOOGLE_SERVICE_ACCOUNT_JSON_TEMP_FILE_LOCATION;
+import static com.amazonaws.athena.connectors.gcs.GcsConstants.SSL_CERT_FILE_LOCATION;
+import static java.util.Objects.requireNonNull;
+
+public class GcsUtil
+{
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+    static List<String> supportedTypeList = Arrays.asList("parquet", "csv");
+
+    private GcsUtil()
+    {
+    }
+
+    /**
+     * Builds a string representation of an instance of {@link StorageSplit}
+     *
+     * @param splits An instance of {@link StorageSplit}
+     * @return String representation of an instance of {@link StorageSplit}
+     * @throws JsonProcessingException If JSON processing error happens
+     */
+    public static synchronized String splitAsJson(List<StorageSplit> splits) throws JsonProcessingException
+    {
+        return objectMapper.writeValueAsString(splits);
+    }
+
+    public static boolean isFieldTypeNull(Field field)
+    {
+        return field.getType() == null
+                || field.getType().equals(Types.MinorType.NULL.getType());
+    }
+
+    /**
+     * Install cacert from resource folder to temp location
+     * This is required for dataset api
+     */
+    public static void installCaCertificate() throws IOException
+    {
+        ClassLoader classLoader = GcsRecordHandler.class.getClassLoader();
+        File file = new File(requireNonNull(classLoader.getResource("")).getFile());
+        File src = new File(file.getAbsolutePath() + File.separator + "cacert.pem");
+        File destination = new File(System.getenv(SSL_CERT_FILE_LOCATION));
+        File parentDestination = new File(destination.getParent());
+        if (!destination.exists()) {
```

Comment:
from description 'This is required for dataset api'.

What if the user override parameter from .yaml file below to some random value. This will failed to copy. Should we throw exception for that case?

```
     SSL_CERT_FILE: '/tmp/cacert.pem'
```

Reply:
We shouldn't throw an exception ourselves.

The right thing to do here if we want to error out is to just not perform any checks.

Just perform the Files.copy() and let Files.copy() itself throw if it can't do it.